app/scripts/modules/ecs/src/serverGroup/configure/wizard/container/Container.tsx

```diff
@@ -152,13 +155,19 @@ export class Container extends React.Component<IContainerProps, IContainerState>
     this.setState({ targetGroupMappings: currentMappings });
   };
 
+  private updateDirtyTargetGroups = () => {
+    this.props.command.viewState.dirty.targetGroups = null;
+    this.setState({dirtyTargetGroups: []});
+  };
```

Comment:
Not sure I agree with marking the invalidness of a target group as part of the state. this property should be determined using the backing data, and be updated in the configuration when the target group changes.

Reply:
So the reason behind setting this param as a part of the state variables is we need to re-render the react component once user clicks the **okay** button. If we need to hide the error message on **okay** click event then we need to save this variable as a part of the state. Also, we are changing the dirty.targetGroups property in the configurations when target group changes and not in the react component.

Reply:
We shouldn't be hiding this message. If they have a target group that's no longer available, the only way they should be able to remove the message should be to change the target group. Removing the message just gives a false sense of security that everything is okay, when in reality, keeping an invalid target group will just cause the deployment to fail.